ci-operator/step-registry/ipi/conf/vsphere/windows-machineset/ipi-conf-vsphere-windows-machineset-commands.sh

```diff
@@ -0,0 +1,43 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+# Wait for WMCO to be up and running
+oc wait deployment windows-machine-config-operator -n openshift-windows-machine-config-operator --for condition=Available=True --timeout=5m
+
+# Ensure userDataSecret exist, fail otherwise. The userDataSecret is required and contains specific information to
+# customize the machine at first boot. For instance, the authorized public key for the SSH server to accept
+# incoming connections, firewall rules, etc.
+timeout 3m bash -c 'until oc -n openshift-machine-api get secret "${WINDOWS_USER_DATA_SECRET}" 2> /dev/null; do echo -n "." && sleep 15; done'
+
+# Get machineset name to generate a generic template
+ref_machineset_name=$(oc -n openshift-machine-api get -o 'jsonpath={range .items[*]}{.metadata.name}{"\n"}{end}' machinesets | grep worker | head -n1)
+
+# Replace machine name `worker` with `winworker`
+winworker_machineset_name="${ref_machineset_name/worker/winworker}"
```

Comment:
You might run into issues with the hostname [restrictions](https://learn.microsoft.com/en-us/troubleshoot/windows-server/identity/naming-conventions-for-computer-domain-site-ou). I suggest using just `winworker`.

Reply:
Thanks @aravindhp for review , we have used similar in aws and gcp windows-machineset as well , do you think it might be needed to modify there as well ? 
cc @rrasouli

Reply:
from the doc you shared , it certainly does , just want to make sure in that case will fix this along with them .

Reply:
In GCP and AWS, WMCO does not override the hostname, so you should be fine. But Azure, Nutanix, vSphjere it is safer to use short names.